Building An SIR Model.rst

```diff
@@ -0,0 +1,29 @@
+
+
+
+Introduction
+------------
+
+Bharatsim is an open-source large-scale agent-based simulation framework that allows you to build and simulate sophisticated agent-based models. Agents form the main component of such a simulation. They are entities that are defined with a set of parameters and obey certain specified conditions as the simulation runs.
+
+In the purview of the SIR model, the agents are people, which go about their daily lives and follow regular schedules. Here, the parameters defining an agent are a unique agent id, their age, their infection status, the time for which they remain infected and so on. The framework allows you to introduce new parameters as and when required.
+
+The rules governing the behaviour of all agents include the particular schedules which they follow and the functions which allow the agents to transit from one compartment to another as described before.
+
+Model Components
+----------------
+
+The model consists of different objects and classes and can be separated into individual ``.scala`` files.
+
+1. Nodes of the Network: These include ``House``, ``School``, ``Office``, etc, which essentially correspond to the different geographical locations in which the agents spend their time according to some schedule. These classes have some attributes associated with them. For eg. ``getContactProbability`` is one such function that gives the transmission probability of the disease in that particular location. In order to add new Nodes to the Network (for eg. Hospital), one has to create a new scala class inside the same package corresponding to the new Node.
+
+2. Disease: ``Disease`` is a class where all epidemiological parameters are conveniently defined. For example, the transmission rate :math:`\beta` is defined in this class. This class allows for a convenient way to store and modify variables as required.
+
+3. Infection Status: The ``InfectionStatus`` is the class where all the compartments of the model are defined. The ``InfectionStatus`` object in the class, is a serialisable object which includes all the compartments. Inside this class, two functions act as encoders(converting the compartment into a corresponding string) and the decoder, which decodes a string into a particular compartment. To add a new compartment like ‘Exposed’ refer to the figure below:-

```

Comment:
Don't talk about a "compartment", this is more a new "infection status", rather than a compartment. If possible, link to some external documentation on enums or something here, people who don't have familiarity with java might have difficulty with this.

Reply:
So I think we are talking about compartments in the Epidemiology section so people become familiar with that terminology.